b2sdk/transfer/emerge/executor.py

```diff
@@ -315,15 +359,25 @@ def _match_unfinished_file_if_possible(
 
         This is only possible if the application key being used allows ``listFiles`` access.
         """
+        file_retention = file_retention or NO_RETENTION_FILE_SETTING
         for file_ in self.services.large_file.list_unfinished_large_files(
             bucket_id, prefix=file_name
         ):
             if file_.file_name != file_name:
                 continue
             if file_.file_info != file_info:
                 continue
+            # FIXME: what if `encryption is None` - match ANY encryption? :)
             if encryption is not None and encryption != file_.encryption:
                 continue
+            if bool(legal_hold) != file_.legal_hold:
+                # when `file_.legal_hold is None` it means that `legal_hold` is unknown and we skip
```

Comment:
shouldn't it then be `if file_.legal_hold is None or legal_hold != file_.legal_hold` ?